boards/cats_rev1Pro/Core/Src/tasks/task_baro_read.c

```diff
@@ -30,33 +32,38 @@ static void read_baro();
 void task_baro_read(void *argument) {
   /* For periodic update */
   uint32_t tick_count, tick_update;
+  uint32_t stage = READ_BARO_TEMPERATURE;
   /* actual measurements from sensor */
   int32_t temperature[3];
   int32_t pressure[3];
 
   tick_count = osKernelGetTickCount();
-  tick_update = osKernelGetTickFreq() / CONTROL_SAMPLING_FREQ;
-
+  tick_update = osKernelGetTickFreq() / (2*CONTROL_SAMPLING_FREQ);
+  prepare_temp();
```

Comment:
why do we prepare_temp() and then we call read_baro() on line :49?

Reply:
The function names are a bit weird actually, can't we say: read_baro(TEMP), read_baro(PRESSURE) or something like that?

Reply:
When calling `read_baro()` we read out the prepared register, it can either be pressure or temperature. 
The process is as follows:
1. Read out prepared data
2. Request new data

Reply:
When calling read_baro() we read out the prepared register, it can either be pressure or temperature.
The process is as follows:

1. Read out prepared data
2. Request new data